image/docker_schema2.go

```diff
@@ -28,33 +28,33 @@ var gzippedEmptyLayer = []byte{
 // gzippedEmptyLayerDigest is a digest of gzippedEmptyLayer
 const gzippedEmptyLayerDigest = digest.Digest("sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4")
 
-type descriptor struct {
+type Descriptor struct {
 	MediaType string        `json:"mediaType"`
 	Size      int64         `json:"size"`
 	Digest    digest.Digest `json:"digest"`
 	URLs      []string      `json:"urls,omitempty"`
 }
 
-type manifestSchema2 struct {
+type ManifestSchema2 struct {
```

Comment:
Having a single schema2 data structure in the `manifest` package, used by all of `docker/daemon`, `docker_schema1.go`, `docker_schema2.go`, and the autodetection in `manifest/manifest.go`, does sound great.